src/data.rs

```diff
@@ -283,6 +283,27 @@ pub trait Data<'a> {
         Ok(unsafe { DataNodeRef::from_raw(self.tree(), rnode as *mut _) })
     }
 
+    /// Search in the given data for a single node matching the provided XPath and
+    /// is an RPC/ output nodes or in input nodes.
```

Comment:
The changes look good to me (thanks for the unit tests!). The only issue is this comment line, it should read: `is an RPC/action output node`.